features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/resources/ResourceDataStoreManagerTest.kt

```diff
@@ -0,0 +1,411 @@
+                key = eq(DATASTORE_HASHES_ENTRY_NAME),
+                data = any(),
+                version = anyOrNull(),
+                callback = any(),
+                serializer = eq(mockResourceHashesEntrySerializer)
+            )
+        ).doAnswer {
+            val callback = it.arguments[3] as DataStoreWriteCallback
+            callback.onSuccess()
+        }
+    }
+
+    @Test
+    fun `M return false W isPreviouslySentResource() { resource was not already sent }`() {
+        // Given
+        testedDataStoreManager = ResourceDataStoreManager(
+            featureSdkCore = mockFeatureSdkCore,
+            resourceHashesSerializer = mockResourceHashesEntrySerializer,
+            resourceHashesDeserializer = mockResourceHashesEntryDeserializer
+        )
+
+        // When
+        val wasSent = testedDataStoreManager.isPreviouslySentResource(fakeHash)
+
+        // Then
+        assertThat(wasSent).isFalse()
+    }
+
+    @Test
+    fun `M return true W isPreviouslySentResource() { resource was already sent }`() {
+        // Given
+        testedDataStoreManager = ResourceDataStoreManager(
+            featureSdkCore = mockFeatureSdkCore,
+            resourceHashesSerializer = mockResourceHashesEntrySerializer,
+            resourceHashesDeserializer = mockResourceHashesEntryDeserializer
+        )
+        testedDataStoreManager.cacheResourceHash(fakeHash)
+
+        // When
+        val wasSent = testedDataStoreManager.isPreviouslySentResource(fakeHash)
+
+        // Then
+        assertThat(wasSent).isTrue()
+    }
+
+    @Test
+    fun `M write hash to datastore W cacheResourceHash()`() {
+        // Given
+        testedDataStoreManager = ResourceDataStoreManager(
+            featureSdkCore = mockFeatureSdkCore,
+            resourceHashesSerializer = mockResourceHashesEntrySerializer,
+            resourceHashesDeserializer = mockResourceHashesEntryDeserializer
+        )
+
+        // When
+        testedDataStoreManager.cacheResourceHash(fakeHash)
+
+        // Then
+        verify(mockFeatureScope.dataStore).setValue(
+            key = eq(DATASTORE_HASHES_ENTRY_NAME),
+            data = any(),
+            version = anyOrNull(),
+            callback = any(),
+            serializer = eq(mockResourceHashesEntrySerializer)
+        )
+    }
+
+    @Test
+    fun `M be threadsafe W cacheResourceHash() { store on one thread, read on another }`() {
+        // Given
+        testedDataStoreManager = ResourceDataStoreManager(
+            featureSdkCore = mockFeatureSdkCore,
+            resourceHashesSerializer = mockResourceHashesEntrySerializer,
+            resourceHashesDeserializer = mockResourceHashesEntryDeserializer
+        )
+        val writeThread = Runnable {
+            testedDataStoreManager.cacheResourceHash(fakeHash)
+        }
+
+        val readThread = Runnable {
+            assertThat(testedDataStoreManager.isPreviouslySentResource(fakeHash))
+                .isTrue()
+        }
```

Comment:
these are not 2 different threads, this is still the same thread. `Runnable` is a simple interface representing unit of work.

I'm not sure that it is easy to write a proper test for the assertion of the thread-safety here, so let's delete this test, in the current state it doesn't correspond to what it wants to do anyway.